cpp/src/arrow/util/small_vector.h

```diff
@@ -26,6 +26,7 @@
 #include <new>
 #include <type_traits>
 #include <utility>
+#include <vector>
```

Comment:
I'm curious why this is necessary. Is `std::vector` used in this header?

Reply:
Yes, in the conversion. Perhaps a forward declaration could suffice? I'm not sure given the definition of the function is also present (not an expert).

https://github.com/apache/arrow/blob/0b3cae41b7346130833e89c3272b526a33387978/cpp/src/arrow/util/small_vector.h#L258

Reply:
Ah, so let's keep the inclusion then. It's almost everywhere in Arrow C++ anyway.